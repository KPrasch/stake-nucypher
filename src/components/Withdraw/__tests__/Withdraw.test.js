import React from 'react';
import Withdraw from '../Withdraw';
import { render, fireEvent, wait } from '@testing-library/react'

describe('<Withdraw />', () => {

  it('should render correctly', () => {
    const { asFragment } = render(<Withdraw />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should trigger onWithdraw on withdrawal', async () => {
    const onWithdraw = jest.fn(() => {});
    const { getByLabelText, getByText } = render(<Withdraw onWithdraw={onWithdraw} />);

    await wait(() => {
      fireEvent.change(getByLabelText('Amount'), { target: { value: '15000' } });
      fireEvent.click(getByText('Withdraw'));
    });

    expect(onWithdraw).toBeCalledWith(expect.objectContaining({ withdrawValue: '15000' }));
  });

  it('should validate input from user correctly', async () => {
    const { getByLabelText, getByText, asFragment } = render(<Withdraw />);

    await wait(() => {
      fireEvent.change(getByLabelText('Amount'), { target: { value: '' } });
      fireEvent.click(getByText('Withdraw'));
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
