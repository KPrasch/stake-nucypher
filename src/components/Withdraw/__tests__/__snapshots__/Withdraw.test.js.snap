// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Withdraw /> should render correctly 1`] = `
<DocumentFragment>
  <form
    class="w-100"
  >
    <div
      class="form-group row"
    >
      <label
        class="form-label col-form-label col-md-2 offset-md-2"
        for="withdraw-value"
      >
        Amount
      </label>
      <div
        class="col-md-4"
      >
        <input
          class="form-control form-control-bg"
          id="withdraw-value"
          name="withdrawValue"
          type="text"
          value="0"
        />
        <div
          class="invalid-feedback"
          style="display: none;"
        />
      </div>
      <div
        class="col-md-2"
      >
        <button
          class="btn btn-link btn-sm"
          type="button"
        >
          Set max
        </button>
      </div>
    </div>
    <div
      class="d-flex justify-content-center"
    >
      <button
        class="btn btn-primary"
        type="submit"
      >
        Withdraw
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`<Withdraw /> should validate input from user correctly 1`] = `
<DocumentFragment>
  <form
    class="w-100"
  >
    <div
      class="form-group row"
    >
      <label
        class="form-label col-form-label col-md-2 offset-md-2"
        for="withdraw-value"
      >
        Amount
      </label>
      <div
        class="col-md-4"
      >
        <input
          class="form-control form-control-bg"
          id="withdraw-value"
          name="withdrawValue"
          type="text"
          value=""
        />
        <div
          class="invalid-feedback"
          style="display: inline;"
        >
          Field is required.
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <button
          class="btn btn-link btn-sm"
          type="button"
        >
          Set max
        </button>
      </div>
    </div>
    <div
      class="d-flex justify-content-center"
    >
      <button
        class="btn btn-primary"
        disabled=""
        type="submit"
      >
        Withdraw
      </button>
    </div>
  </form>
</DocumentFragment>
`;
